### PR TITLE
fix(tree-view): add aria multiselectable attribute when needed

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
`clr-tree` throws ExpressionChangedAfterItHasBeenCheckedError when setting `aria-multiselectable`.

Issue Number: #1562, CDE-2301

## What is the new behavior?
Aria multiselectable attribute is added when needed.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/data/tree-view/tree.ts [hunk 1]
```diff
@@ -35,30 +35,30 @@ import { ClrTreeNode } from './tree-node';
...
 export class ClrTree<T> implements AfterContentInit, OnDestroy {
   @ContentChildren(ClrTreeNode) private rootNodes: QueryList<ClrTreeNode<T>>;
 
   private subscriptions: Subscription[] = [];
+  private _isMultiSelectable = false;
 
   constructor(
     public featuresService: TreeFeaturesService<T>,
     private focusManagerService: TreeFocusManagerService<T>,
-    { nativeElement }: ElementRef<HTMLElement>,
```
Comment (mivaylo) [inline projects/angular/src/data/tree-view/tree.ts:49]: I would just add a `private` modifier and thus use the nativeElement throught the class as a field with less changes but I am fine both ways

**Review by mivaylo — APPROVED** [review]

File: projects/angular/src/data/tree-view/tree.ts (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/tree-view/tree.ts:49]: It won't work. I tested it.

File: projects/angular/src/data/tree-view/tree.ts [hunk 2]
```diff
@@ -35,30 +35,30 @@ import { ClrTreeNode } from './tree-node';
...
-    renderer: Renderer2,
+    private renderer: Renderer2,
+    private el: ElementRef<HTMLElement>,
     ngZone: NgZone
   ) {
     const subscription = ngZone.runOutsideAngular(() =>
-      fromEvent(nativeElement, 'focusin').subscribe((event: FocusEvent) => {
-        if (event.target === nativeElement) {
+      fromEvent(this.el.nativeElement, 'focusin').subscribe((event: FocusEvent) => {
```
Comment (kevinbuhmann) [inline projects/angular/src/data/tree-view/tree.ts:54]:
```suggestion
      fromEvent(el.nativeElement, 'focusin').subscribe((event: FocusEvent) => {
```

The `this` keyword is not needed in the constructor.

File: projects/angular/src/data/tree-view/tree.ts [hunk 3]
```diff
@@ -35,30 +35,30 @@ import { ClrTreeNode } from './tree-node';
...
+        if (event.target === this.el.nativeElement) {
```
Comment (kevinbuhmann) [inline projects/angular/src/data/tree-view/tree.ts:55]:
```suggestion
        if (event.target === el.nativeElement) {
```

File: projects/angular/src/data/tree-view/tree.ts (hunk 2 above)
Comment (kevinbuhmann) [reply projects/angular/src/data/tree-view/tree.ts:54]: Please update all usages in the constructor.

Comment (valentin-mladenov) [reply projects/angular/src/data/tree-view/tree.ts:54]: We can, but in this specific constructor we use `this` for the `focusManagerService`.
Also we use `this` in every constructor for `private` or `public` fields.

Comment (kevinbuhmann) [reply projects/angular/src/data/tree-view/tree.ts:54]: Okay, that's fine.

File: projects/angular/src/data/tree-view/tree.ts [hunk 4]
```diff
@@ -72,13 +72,21 @@ export class ClrTree<T> implements AfterContentInit, OnDestroy {
...
     this.setRootNodes();
     this.subscriptions.push(
       this.rootNodes.changes.subscribe(() => {
+        if (this.featuresService.selectable && this.rootNodes.length > 0) {
+          this._isMultiSelectable = true;
+          this.renderer.setAttribute(this.el.nativeElement, 'aria-multiselectable', 'true');
+        } else {
+          this._isMultiSelectable = false;
+          this.renderer.removeAttribute(this.el.nativeElement, 'aria-multiselectable');
+        }
```
Comment (kevinbuhmann) [inline projects/angular/src/data/tree-view/tree.ts:88]: I would extract this to a `setMultiSelectable` method.

File: projects/angular/src/data/tree-view/tree.ts (hunk 2 above)
Comment (kevinbuhmann) [reply projects/angular/src/data/tree-view/tree.ts:54]: I opened a separate PR to clean this up everywhere. #1566

File: projects/angular/src/data/tree-view/tree.ts (hunk 4 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/tree-view/tree.ts]: corrected in ca05de4

File: projects/angular/src/data/tree-view/tree.ts (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/data/tree-view/tree.ts:49]: @mivaylo should we resolve this comment?

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**